test/connectors/AuthConnectorSpec.scala

```diff
@@ -0,0 +1,253 @@
+        )thenReturn(httpResponse.status,headers,httpResponse.body)
+
+       await(connector.login(authUserDetails)).value shouldBe Left("Internal Error, missing headers or gatewayToken in response from auth-login-api")
+
+      }
+
+
+      "Error for missing location header" in {
+
+        val authUserDetails = AuthUserDetails.empty()
+        val token : String = "token13579"
+
+
+        val headers = Map(HeaderNames.AUTHORIZATION -> "auth1")
+        val httpResponse = HttpResponse(201, Json.parse(s"""{"gatewayToken":"$token"}"""), Map.empty[String, Seq[String]])
+
+        when(
+          POST,
+          "/government-gateway/session/login"
+        )thenReturn(httpResponse.status,headers,httpResponse.body)
+
+        await(connector.login(authUserDetails)).value shouldBe Left("Internal Error, missing headers or gatewayToken in response from auth-login-api")
+
+      }
+
+      "Error for missing gateway token" in {
+
+        val authUserDetails = AuthUserDetails.empty()
+
+
+        val headers = Map(HeaderNames.AUTHORIZATION -> "auth1",HeaderNames.LOCATION -> "loc1")
+        val httpResponse = HttpResponse(201, emptyJsonBody, Map.empty[String, Seq[String]])
+
+        when(
+          POST,
+          "/government-gateway/session/login"
+        )thenReturn(httpResponse.status,headers,httpResponse.body)
+
+        await(connector.login(authUserDetails)).value shouldBe Left("Internal Error, missing headers or gatewayToken in response from auth-login-api")
+
+      }
+
+      "verify error status codes" in {
+        List(
+          HttpResponse(400, emptyJsonBody),
+          HttpResponse(401, emptyJsonBody),
+          HttpResponse(403, emptyJsonBody),
+          HttpResponse(500, emptyJsonBody),
+          HttpResponse(502, emptyJsonBody),
+          HttpResponse(503, emptyJsonBody)
+        ).foreach { httpResponse =>
+          val authUserDetails = AuthUserDetails.empty()
+
+          val headers = Map(HeaderNames.AUTHORIZATION -> "auth1", HeaderNames.LOCATION -> "loc1")
+
+          when(
+            POST,
+            "/government-gateway/session/login"
+          )thenReturn(httpResponse.status,headers,httpResponse.body)
+
+          await(connector.login(authUserDetails)).value shouldBe Left(s"failed calling auth-login-api, got status ${httpResponse.status}, body: ${httpResponse.body}")
+        }
+      }
+    }
+
+
+    "getPrivilegedToken" must{
+      val privilegedRequestBody: JsValue = JsObject(
+        Map(
+          "clientId"   -> JsString("id"),
+          "enrolments" -> JsArray(),
+          "ttl"        -> JsNumber(1200) // scalastyle:ignore magic.number
+        )
+      )
+
+      "returns token successfully" in {
+
+        val token = "token123"
+
+        val headers = Map(HeaderNames.AUTHORIZATION -> token)
+
+        val httpResponse : HttpResponse = HttpResponse(200, emptyJsonBody, Map.empty[String, Seq[String]])
+
+        when(
+          POST,
+          "/auth/sessions",
+          Map.empty,
```

Comment:
Can use named parameter here for `body`, that way we don't need to pass `Map.empty` for parameters and headers